src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,174 @@
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { getDefaultSlippageDecimalPercentageForSwapper } from 'constants/constants'
+import type { Address } from 'viem'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { getThorTxInfo as getEvmThorTxInfo } from 'lib/swapper/swappers/ThorchainSwapper/evm/utils/getThorTxData'
+import { isFulfilled, isRejected, isResolvedErr } from 'lib/utils'
+import { convertDecimalPercentageToBasisPoints } from 'state/slices/tradeQuoteSlice/utils'
+
+import { ALLOWANCE_CONTRACT } from '../constants'
+import type { ThorTradeQuote } from '../getThorTradeQuote/getTradeQuote'
+import { addAggregatorAndDestinationToMemo } from './addAggregatorAndDestinationToMemo'
+import { getBestAggregator } from './getBestAggregator'
+import { getL1quote } from './getL1quote'
+import type { AggregatorContract } from './longTailHelpers'
+import { getTokenFromAsset, getWrappedToken, TradeType } from './longTailHelpers'
+
+// This just uses UniswapV3 to get the longtail quote for now.
+export const getL1ToLongtailQuote = async (
+  input: GetTradeQuoteInput,
+  streamingInterval: number,
+  assetsById: AssetsByIdPartial,
+): Promise<Result<ThorTradeQuote[], SwapErrorRight>> => {
+  const {
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+    sellAsset,
+    slippageTolerancePercentageDecimal,
+  } = input
+
+  /*
+    We only support L1 -> ethereum longtail swaps for now.
+  */
+  if (buyAsset.chainId !== ethChainId) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - Unsupported chainId ${buyAsset.chainId}.`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { buyAssetChainId: buyAsset.chainId },
+      }),
+    )
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+  const sellChainId = sellAsset.chainId
+  const buyChainId = buyAsset.chainId
+
+  const nativeSellAssetId = chainAdapterManager.get(sellChainId)?.getFeeAssetId()
+  const nativeSellAsset = nativeSellAssetId ? assetsById[nativeSellAssetId] : undefined
+
+  const nativeBuyAssetId = chainAdapterManager.get(buyChainId)?.getFeeAssetId()
+  const nativeBuyAsset = nativeBuyAssetId ? assetsById[nativeBuyAssetId] : undefined
+
+  if (!nativeBuyAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${buyChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { buyAssetChainId: buyChainId },
+      }),
+    )
+  }
+
+  if (!nativeSellAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${sellChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  // TODO: use more than just UniswapV3, and also consider trianglar routes.
```

Comment:
Feels like the TODO above already captures this and this doesn't really do anything programmatic by aggregator in and by itself, can probably be removed